shared/vcjs/verifyCredential.ts

```diff
@@ -26,10 +27,17 @@ const ProofPurpose = {
 
 export async function verifyCredential(
   verifiableCredential: VerifiableCredential | Credential,
+  vcMetadata: VCMetadata,
 ): Promise<VerificationResult> {
   try {
     let purpose: PublicKeyProofPurpose | AssertionProofPurpose;
-    switch (verifiableCredential.proof.proofPurpose) {
+    const proof = VCMetadata.fromVC(vcMetadata).isFromOpenId4VCI()
```

Comment:
can we not add this check "VCMetadata.fromVC(vcMetadata).isFromOpenId4VCI()" here, lets check it before calling and send only the credential, so verifyCredential Method resposibilty is to just verify and not figuring out where the credential would be.

Reply:
Done as part of another PR